core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -80,6 +80,24 @@ public static Type convert(Schema schema) {
     return AvroSchemaVisitor.visit(schema, new SchemaToType(schema));
   }
 
+  static boolean hasIds(Schema schema) {
+    try {
+      // Try to convert the type to Iceberg. If an ID assignment is needed, return false.
+      AvroSchemaVisitor.visit(schema, new SchemaToType(schema) {
+        @Override
+        protected int allocateId() {
```

Comment:
Can we pass this as a lambda instead? Seems like quite a bit of boiler-plate.

Reply:
I ended up not extending `SchemaToType` as it was cumbersome to implement the logic you mentioned. I've written a new visitor instead